### PR TITLE
Do not crash with weird RegExp

### PR DESCRIPTION
**Pull request by mcollina**
Fixes https://github.com/nearform/node-clinic-flame/issues/40

### CODE REVIEW

File: analysis/frame-node.js [hunk 1]
```diff
@@ -169,10 +169,11 @@ class FrameNode {
     const platformPath = getPlatformPath(systemInfo)
 
     const parentDir = platformPath.join(systemInfo.mainDirectory, `..${systemInfo.pathSeparator}`)
+    const typeTEMP = platformPath.relative(parentDir, platformPath.dirname(this.fileName))
 
     return {
       // TODO: use this type after adding custom d3-fg filter on properties including category
-      typeTEMP: platformPath.relative(parentDir, platformPath.dirname(this.fileName)),
```
Comment (AlanSl) [inline analysis/frame-node.js]: This change doesn't seem to do anything? Is it left over from debugging?

Doesn't really matter, but it'll cause a minor merge conflict in at least one open PR that removes typeTEMP

File: test/analysis-categorise.test.js [hunk 2]
```diff
@@ -175,3 +175,12 @@ test('analysis - categorise node properties', (t) => {
 
   t.end()
 })
+
+test('do not crash on RegExp nodes', (t) => {
+  const regexpNode = new FrameNode({
+    name: '\\.js(\\?[^.]+)?$ [CODE:RegExp]'
+  })
+  regexpNode.categorise(windows)
+  t.equal(regexpNode.type, 'regexp')
```
Comment (mcollina) [inline test/analysis-categorise.test.js]: @AlanSl do you think we should add more assertions here?

File: analysis/frame-node.js (hunk 1 above)
Comment (mcollina) [reply analysis/frame-node.js]: Updated.

File: test/analysis-categorise.test.js (hunk 2 above)
Comment (AlanSl) [reply test/analysis-categorise.test.js]: It'd be good to have one that ensures that a regex containing `/node_modules/` or `\\node_modules\\` doesn't get incorrectly classified as deps.

Besides that, I'd prefer to check all of this logic for edge cases but as a separate piece of work, so we can release this fix quickly.

**Comment by mcollina** [comment]
PTAL

**Comment by AlanSl** [comment]
LGTM. Let's release this then I'll do the follow-up card on Sunday.

**Pull request merged**